comp_sci_gender_bias/pipeline/urn_to_school_name_lookup/urn_to_school_name_lookup.py

```diff
@@ -0,0 +1,56 @@
+from comp_sci_gender_bias import PROJECT_DIR
+from comp_sci_gender_bias.getters.school_data import school_table
+from comp_sci_gender_bias.getters.dfe_combined_school_data import (
+    dfe_combined_school_data,
+)
+from comp_sci_gender_bias.utils.process_pandas import remove_nonalphanum_lowercase
+import pandas as pd
+
+URN_SCHOOL_LOOKUPS_PATH = PROJECT_DIR / "inputs/data/urn_school_lookups"
+SAVE_FULL_LOOKUP_PATH = URN_SCHOOL_LOOKUPS_PATH / "urn_school_lookup_full.csv"
+MANUAL_LOOKUP_PATH = URN_SCHOOL_LOOKUPS_PATH / "urn_school_lookup_manual.csv"
+
+
+def make_auto_lookup() -> pd.DataFrame:
+    """Produce a unique reference number to school name lookup by matching
+    between the combined Department for Education dataset and the school master table.
+    There is some simple processing performed on the school names to enable more matches"""
```

Comment:
Maybe you tried this already or the gains would be marginal, but calculating the levenshtein distance might be useful if there are still some almost-matches at the end of the text cleaning and standardisation.

Reply:
I didn't try that as the amount of schools that weren't matching was small and some weren't matching because the name had changed so I just did those matches manually.